File: src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerStyles.java
package com.vaadin.demo.component.datetimepicker;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.datetimepicker.DateTimePicker;
import com.vaadin.flow.component.datetimepicker.DateTimePickerVariant;
import com.vaadin.flow.router.Route;

import java.time.LocalDateTime;

import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("date-time-picker-styles")
public class DateTimePickerStyles extends HorizontalLayout {

    public DateTimePickerStyles() {
        setPadding(false);

        // tag::snippet[]
        DateTimePicker field = new DateTimePicker();
        field.addThemeVariants(
            DateTimePickerVariant.LUMO_SMALL,
            DateTimePickerVariant.LUMO_ALIGN_RIGHT,
            DateTimePickerVariant.LUMO_HELPER_ABOVE_FIELD
        );
        field.getStyle().set("--vaadin-input-field-border-width", "1px");
        // end::snippet[]
        field.setLabel("Label");
        field.setHelperText("Helper text");
        field.setValue(LocalDateTime.of(2020, 6, 12, 12, 30));

        add(field);
    }

    public static class Exporter extends DemoExporter<DateTimePickerStyles> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/datetimepicker/date-time-picker-styles.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/date-time-picker';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('date-time-picker-styles')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
        <!-- tag::snippet[] -->
        <vaadin-date-time-picker theme="align-right small helper-above-field"
          label="Label" helper-text="Helper text" value="2020-06-12T12:30"
          style="--vaadin-input-field-border-width: 1px;">
        </vaadin-date-time-picker>
        <!-- end::snippet[] -->
    `;
  }
}
